runtime/compiler/control/rossa.cpp

```diff
@@ -1564,11 +1566,10 @@ onLoadInternal(
 
    // If more than one diagnostic compilation thread is created, MAX_DIAGNOSTIC_COMP_THREADS needs to be updated
    // create diagnostic compilation thread
-   if (compInfo->startCompilationThread(-1, highestThreadID, /* isDiagnosticThread */ true) != 0)
+   if (compInfo->startCompilationThread(-1, compInfo->getFirstDiagThreadID(), /* isDiagnosticThread */ true) != 0)
```

Comment:
The code here and in `startCompilationThread()` assumes that diagnostic threads comes immediately after usable compilation threads. This contradicts the message for the second commit: "Rather than assuming that the first n IDs are compilation threads
followed by the diagnostic thread, use APIs from the previous commits to
determine the thread ID ranges."

Reply:
That message was mainly for all places that iterate through the threads. In rossa.cpp, the threads are initialized and so to maintain the previous behaviour, it's initialized such that the first n thread ids are comp threads, followed by the diagnostic thread. I can move this bit of code into its own commit.